subsys/bluetooth/mesh/main.c

```diff
@@ -369,10 +354,17 @@ int bt_mesh_start(void)
 		bt_mesh_beacon_disable();
 	}
 
-	if (IS_ENABLED(CONFIG_BT_MESH_GATT_PROXY) &&
-	    bt_mesh_gatt_proxy_get() != BT_MESH_GATT_PROXY_NOT_SUPPORTED) {
-		bt_mesh_proxy_gatt_enable();
-		bt_mesh_adv_update();
+	/* For PB-GATT provision, will enable in le disconnect handler. */
+	if (bt_mesh_prov_link.bearer->type == BT_MESH_PROV_ADV) {
+		if (IS_ENABLED(CONFIG_BT_MESH_PB_GATT)) {
+			(void)bt_mesh_pb_gatt_disable();
```

Comment:
If PB-GATT is disabled here, then PB-GATT service is unregistered before the provisioner disconnects from PB-GATT:
Provisioning Data PDU received -> `prov_device.c:prov_data` -> `bt_mesh_provision` -> `bt_mesh_start` -> `bt_mesh_pb_gatt_disable` -> `is_register = false`. Then later `pb_gatt.c:gatt_disconnect` gets called, but the all is dropped because `is_register` is false.